### ISSUE TITLE
Maintain release notes for corefx NuGet packages

### ISSUE THREAD

**Issue by madelson** [dotnet/corefx#23570]
With corefx, it's great that development happens in the open and that pull requests are accepted. However, it is hard to know when various features or fixes actually make it out because the packages on NuGet.org do not specify release notes. For example, System.Collections.Immutable links it's release notes as https://github.com/dotnet/core/blob/master/release-notes/1.1/1.1.0-preview1.md, which isn't very helpful.

It would be great if each package linked to a clear release notes page with a bulleted list of fixes and features linked to issues. A good example of a release notes page is https://github.com/JamesNK/Newtonsoft.Json/releases.

**Comment by karelz** [comment]
@madelson are you looking for something like list of all issues and/or PRs in a particular milestone? E.g. for 2.0.0: https://github.com/dotnet/corefx/milestone/4 (8.5K entries) ... that may be easily added to release notes, except that it spans multiple repos, so there will be multiple links. Also, I am not sure if other repos use milestones as rigorously as I do on CoreFX repo. Another option is just to point to commit list in each repo - rather long list.
Is that what you're looking for?

cc @richlander @leecow @terrajobst

**Comment by madelson** [comment]
@karelz yes linking to a page like in the NuSpec release notes (so that you can easily find it from NuGet.org) would be a good start.

Ideally, the release notes bring you to a page that is a bit more curated in the sense that it highlights the most impactful changes.

It is also nice to be able to scroll through the notes from several versions.

Essentially, the use-case is that in considering updating any package you want to understand what has changed between the version you currently reference and other versions (sometimes looking for a specific update). With CoreFx, this is difficult to do since you have to map between NuGet versions and GitHub milestones.

**Comment by karelz** [comment]
Not sure what you mean by NuSpec.

Based on what you expect, I think that a link to https://github.com/dotnet/core/blob/master/release-notes/2.0/2.0.0.md would be appropriate solution - all the info (incl. list of commits) is there. Or do you expect more than that?

**Comment by danmosemsft** [comment]
@karelz the content shown in the nuget/myget page for our packages is partly generated from the .nuspec file in the package. 

Eg., on https://www.nuget.org/packages/Microsoft.NETCore.App/ the text `A set of .NET API's that are included in the default .NET Core application model. 
e8b8861ac7faf042c87a5c2f9f2d04c98b69f28d 
When using NuGet 3.x this package requires at least version 3.4.` is from the .nuspec file we control and we could put more text there.
